x/commitment/keeper/total_supply.go

```diff
@@ -0,0 +1,60 @@
+package keeper
+
+import (
+	"cosmossdk.io/math"
+	storetypes "cosmossdk.io/store/types"
+	"github.com/cosmos/cosmos-sdk/runtime"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/elys-network/elys/x/commitment/types"
+	ptypes "github.com/elys-network/elys/x/parameter/types"
+)
+
+func (k Keeper) GetTotalSupply(ctx sdk.Context) (val types.TotalSupply) {
+	store := runtime.KVStoreAdapter(k.storeService.OpenKVStore(ctx))
+	b := store.Get(types.TotalSupplyKeyPrefix)
+
+	if b != nil {
+		k.cdc.MustUnmarshal(b, &val)
+	} else {
+		val.TotalEdenSupply = math.ZeroInt()
+		val.TotalEdenbSupply = math.ZeroInt()
+	}
+	return
+}
+
+func (k Keeper) SetTotalSupply(ctx sdk.Context, val types.TotalSupply) {
+	store := runtime.KVStoreAdapter(k.storeService.OpenKVStore(ctx))
+	b := k.cdc.MustMarshal(&val)
+	store.Set(types.TotalSupplyKeyPrefix, b)
+}
+
+func (k Keeper) V9_SetSupply(ctx sdk.Context) {
```

Comment:
There might be some Eden which will already be converted to Elys, we need to add that as well.

Reply:
currently we are tracking total supply in DS, should I add one more variable to track total minted ?